data_extractor/README.rst

```diff
@@ -29,3 +29,55 @@ be displayed after a while (depending on the amount of processing required and
 
 .. _Poetry: https://python-poetry.org/
+
+--------------
+Google Semantic Location History
+--------------
+In this example, we first create simulated Google Semantic Location History (GSLH) data and then extract relevant information from the simulated data.
+
+Data simulation
+--------------
+Command:
+``poetry run python google_semantic_location_history/simulation_gslh.py``
+
+This generates fake GSLH data using the python libraries [GenSON](https://pypi.org/project/genson/), [Faker](https://github.com/joke2k/faker), and [faker-schema](https://pypi.org/project/faker-schema/).
+
+First, generate a JSON schema from a JSON object using ``GenSON's SchemaBuilder`` class. The JSON object is derived from an example GSLH data package, downloaded from Google Takeout. The GSLH data package consists of monthly JSON files with information on e.g. geolocations, addresses and time spend in places and in activity. The JSON schema describes the format of the JSON files, and can be used to generate fake data with the same format. This is done by converting the JSON schema to a custom schema expected by ``faker-schema`` in the form of a dictionary, where the keys are field names and the values are the types of the fields. The values represent available data types in `Faker`, packed in so-called providers. ``Faker`` provides a wide variety of data types via providers, for example for names, addresses, and geographical data. This allows us to easily customize the faked data to our specifications.
+
+This creates a zipfile with the simulated Google Semantic Location History data in ``tests/data/Location History.zip``.
+
+Data extraction
+--------------
+Command:
+``poetry run python google_semantic_location_history/main.py``
+
+This extracts and displays the relevant data and summary. It calls the same ``process`` function as the web application would. To use the GSLH data extraction script in the web application, one needs to specify this in ``pyworker.js``.
+
+--------------
+Google Search History
+--------------
+
+Here, we extract relecvant information from simulated Google Search History (GSH).
+
+Data simulation
+--------------
+Command:
+``poetry run python google_search_history/simulation_gsh.py``
+
+This will generate a dummy Google Takeout folder, containing a simulated BrowserHistory.py file. This file mainly describes when which websites were visited by the user. To simulate (news) web pages, you can either base them on real URLs (see `tests/data/urldata.csv`) or create entirely fake ones using [Faker](https://github.com/joke2k/faker). The timetamp of each web visit is set to be before, during, or after a certain period (in this case, the Dutch COVID-19 related curfew), and is randomly generated using Faker.
+
+Running the code will create a Google Search History file in the same path structure as it would occur when actually downloading a Google Takeout data download package: ``tests/data/takeout.zip/Takeout/Chrome/BrowserHistory.json``.
+
```

Comment:
Running this in my virtual environment causes the zipfile to be located in:
`Created BrowserHistory.json in .venv/lib/python3.8/site-packages/pkg_resources/tests/data/Takeout.zip`
Also a zipfile is not a directory, as the above notation would suggest. Aand is it `takeout.zip` is or `Takeout.zip`?

Reply:
Solved it